### PR TITLE
Support for self-managed DBMS sources of GDS Sessions (take 3)

### PR DESCRIPTION
**Pull request by FlorentinD**
enable creating sessions against stand-alone

Co-Authored-By: Rafal Skolasinski <rafal.skolasinski@neo4j.com>

### CODE REVIEW

**Comment by FlorentinD** [comment]
[[🍓] Enable client to create sessions against self-managed dbms](https://trello.com/c/8FRF05Jk/9295-%F0%9F%8D%93-enable-client-to-create-sessions-against-self-managed-dbms)

File: graphdatascience/session/__init__.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "GdsSessions",
     "SessionInfo",
     "DbmsConnectionInfo",
+    "CloudLocation",
```
Comment (FlorentinD) [inline graphdatascience/session/__init__.py:11]: lets make sure we have API ref as well

File: graphdatascience/session/gds_sessions.py [hunk 2]
```diff
@@ -70,7 +71,9 @@ def get_or_create(
         self,
         session_name: str,
         memory: SessionMemory,
+        # TODO support TTL and cloud_location
         db_connection: DbmsConnectionInfo,
+        cloud_location: Optional[CloudLocation] = None,
```
Comment (FlorentinD) [inline graphdatascience/session/gds_sessions.py:76]: lets add a notebook example (making sure its part of CI but also for user-visibility)

File: graphdatascience/session/__init__.py (hunk 1 above)
Comment (RafalSkolasinski) [reply graphdatascience/session/__init__.py:11]: done

File: graphdatascience/session/gds_sessions.py (hunk 2 above)
Comment (FlorentinD) [reply graphdatascience/session/gds_sessions.py:82]: -- will do in follow up PR

**Pull request merged**